db/repos/transactions.js

```diff
@@ -283,8 +283,15 @@ TransactionsRepo.prototype.save = function (transactions) {
 		batch.push(self.db[self.transactionsRepoMap[type]].save(groupedTransactions[type]));
 	});
 
-	// To avoid nested transactions aka transactions savepoints
-	if(this.db.ctx.isTX) {
+	// In order to avoid nested transactions, aka savepoints, we need to walk up the task chain,
+	// and see if there is any task above us that's a transaction:
+	let insideTransaction, ctx = this.db.ctx;
+	while(ctx && !insideTransaction) {
+		insideTransaction = ctx.isTX;
+		ctx = ctx.parent;
+	}
+
+	if(this.db.ctx && insideTransaction) {
```

Comment:
We are not calling `save` out of transaction in any case, but still its good to have a safeguard here.

Reply:
Yeah, i'm hoping to get to refactor the hell out of it, when its turn comes ;)